CMakeLists.txt

```diff
@@ -20,16 +20,15 @@ cmake_minimum_required (VERSION 2.6)
 
 project (Proton C)
 
+# Enable C++ now for examples and bindings subdirectories, but make it optional.
+enable_language(CXX OPTIONAL)
+
```

Comment:
The code block below already conditionally enables C++, the blocks should be merged